.github/actions/fastlane-android/action.yaml

```diff
@@ -29,35 +30,38 @@ inputs:
     required: false
     default: ""
   java_distribution:
-    description: Option of action/setup-java@v2
+    description: Option of action/setup-java@v4
     required: false
     default: 'adopt'
   java_version:
-    description: Option of action/setup-java@v2
+    description: Option of action/setup-java@v4
     required: false
     default: '11'
 runs:
   using: "composite"
   steps:
-    # Setup ruby on github hosted machines, step fails on self-hosted macjenkins
+    # Setup step fails on self-hosted macjenkins
```

Comment:
why does it fail? 
It should work regardless MAC or Intel?

Reply:
I don't know exactly - this comment and exclusion were added previously, I modified `if` rule to be able to install ruby on self-hosted linux runners, but still skip step on MacOS, since it was claimed as failing.

Reply:
Don't we use a mac machine to build an android? I definitely saw that some android builds took place on the mac, but I'm not sure if this is a prerequisite or not for those projects

Reply:
You can just use previous action version to keep on with mac builds. But I think we must use linux runners whenever possible, because we have only one mac runner which resources must be preserved for ios builds.

Reply:
@populov pls investigate and let's fix why it fails on Mac and adjust, so when/if we need to use mac builders - everything works out of the box.

Reply:
@populov yes, we can build android on m1 runner. I installed Android SDK on it. Example here. I decided do not add this env in github runner configuration, because it is not documented yet.  https://github.com/saritasa-nest/wrdoc-frontend/blob/develop/.github/workflows/ci-dev.yml#L88
```shell
brew install --cask android-commandlinetools
echo $"\nexport ANDROID_HOME=`brew --prefix`/share/android-commandlinetools\n" >> /Users/$(whoami)/.zshrc
# Then Accept all licenses
yes | sdkmanager --licenses
sdkmanager "cmdline-tools;latest" "patcher;v4" "build-tools;30.0.2" "platforms;android-32" "emulator" "tools" "platform-tools"

```

Reply:
@roman-mightydevops Btw, M1 builds android  faster than x86-64 - we compared it on one android project. But I agree with you - it is better built Android on Linux runner and keep macOS clean. 
>But I think we must use linux runners whenever possible